viewing_party/party.py

```diff
@@ -1,23 +1,177 @@
+        return (most_popular[0])
 
-# -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
 
-        
-# -----------------------------------------
-# ------------- WAVE 4 --------------------
-# -----------------------------------------
+def get_unique_watched(user_data):
+    unique = []
+    not_unique = []
+    total_movies_list = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+    total_movies_list = users_movies + friends_movies
+    
+    for item in total_movies_list:
+        if item not in friends_movies:
+            unique.append(item)
+        else:
+            not_unique.append(item)
+    return unique
+
+def get_friends_unique_watched(user_data):
+    unique = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+            
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in unique:
+                unique.append(item)
+    
+    return unique
+# 
+# ------------- WAVE 4 -------------
+
+def get_available_recs(user_data):
+    unique = []
+    users_movies = []
+    friends_movies = []
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+            
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in unique:
+                if item["host"] in user_data["subscriptions"]:
+                    unique.append(item)
+    
+    return unique
 
-# -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    most_genre = []
+    unique = []
+    users_movies = []
+    friends_movies = []
+    
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+        most_genre.append(movie["genre"])
+
+    if len(users_movies) == 0:
+        return friends_movies
+    the_max = max(set(most_genre), key= most_genre.count)
+    for movie in user_data["friends"]:
+        for item in movie["watched"]:
+            friends_movies.append(item)
+            
+    for item in friends_movies:
+        if item not in users_movies:
+            if item not in unique:
+                if item["host"] in user_data["subscriptions"]:
+                    if most_genre[0] in item["genre"]:
+                        unique.append(item)
+    
+    return unique
+
+def get_rec_from_favorites(user_data):

```

Comment:
A lot of the code here is a repeat of finding movies that only the user has watched. We have a function for that! We can use `get_unique_watched`.